tests/e2e/swarm_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite includes test cases to verify basic vDVS functionality
+// in docker swarm mode.
+
+package e2e
+
+import (
+	"log"
+
+	. "gopkg.in/check.v1"
+
+	constant "github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+)
+
+type SwarmTestSuite struct {
+	esxName     string
+	master      string
+	worker1     string
+	worker2     string
+	dockerHosts []string
+	volumeName  string
+	serviceName string
+}
+
+func (s *SwarmTestSuite) SetUpSuite(c *C) {
+	s.esxName = inputparams.GetEsxIP()
+	s.master = inputparams.GetSwarmManager1()
+	s.worker1 = inputparams.GetSwarmWorker1()
+	s.worker2 = inputparams.GetSwarmWorker2()
+	s.dockerHosts = []string{s.master, s.worker1, s.worker2}
+
+	s.volumeName = inputparams.GetVolumeNameWithTimeStamp("swarm_test")
+	s.serviceName = inputparams.GetServiceNameWithTimeStamp("swarm_test")
+
+	// Verify if swarm cluster is already initialized
+	out, err := dockercli.ListNodes(s.master)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Create the volume
+	out, err = dockercli.CreateVolume(s.master, s.volumeName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	status := verification.VerifyDetachedStatus(s.volumeName, s.master, s.esxName)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", s.volumeName))
+}
+
+func (s *SwarmTestSuite) TearDownSuite(c *C) {
+	// Clean up the volume
+	out, err := dockercli.DeleteVolume(s.master, s.volumeName)
+	c.Assert(err, IsNil, Commentf(out))
+}
+
+var _ = Suite(&SwarmTestSuite{})
+
+// Test vDVS usage in swarm mode
+//
+// Test steps:
+// 1. Create a docker service with replicas setting to 1
+// 2. Verify the service is up and running with one node
+// 3. Verify one container is spawned
+// 5. Verify the volume is in attached status
```

Comment:
@lipingxue Can we add this comment in the issue #601 itself? We can discuss your comment in the issue and once we have an agreement we can come back to update this test.